STIPS/STIP-007.md

```diff
@@ -0,0 +1,145 @@
+# STIP-007: Issuance Transformation Pipeline
+*Using template v0.1*
+## Abstract
+This STIP explores an implementation for a transformation pipeline that can take a specified amount of tokens and transform them into all the positions held by a Set for issuance. Conversely, also allow someone to redeem all the positions from a Set into some amount of a specified token.
+## Motivation
+This feature is to be used by issuers and redeemers to pay/receive one asset when issuing/redeeming a SetToken. This "transformation pipeline" will take the one asset and trade it into the necessary positions using a DEX. This feature provides a considerable UX improvment for users over having to replicate every position in a Set in the exact proportions then issuing a Set. Additionally, it does not require seeding an AMM pool with Set liquidity in order to allow users to get into the Set by buying it off of an exchange. The "transformation pipeline" will allow the longer tail of Set managers to enter complex positions without hindering the ability for users to issue or redeem the Set. This feature will however be gas intensive so will only be viable on cheaper EVM compatible L1s and L2s.
+
+This feature should be able to handle the following asset types:
+- Spot tokens
+- Wrapped tokens (AAVE, Compound)
+- LP tokens (Curve, UniswapV2, Sushiswap)
```

Comment:
LP tokens have a special problem of often not having the ability to create a trade with a specified output amount of LP tokens. Instead, they usually require supplying an input amount of the underlying tokens. This setup is incompatible with the issuance modules which would require fixed proportions of each token (including the LP tokens). Have you thought of any potential solutions for this?